### PR TITLE
chore: adding golden files support on webconfig tests

### PR DESCRIPTION
**Pull request by nicolastakashi**
## Description

I'm adding golden file supports on `webconfig` package, solves #5741

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [X] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [ ] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Changelog entry

_Please put a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note
adding golden file supports on `webconfig` package
```

### CODE REVIEW

**Comment by ArthurSens** [comment]
Amazing to see progress here!

One thing that I feel is missing is a Makefile step to update golden files.

 Looking at the [documentation of the library](https://pkg.go.dev/gotest.tools/v3/golden) you've used, if we run `go test -update`, the golden files will be updated automatically. However, running `go test -update` in a package that doesn't have `gotest.tools/v3/golden` fails saying that the flag -update is not expected. 

Let's think of a way of update the golden files without failing everything else 😅

**Comment by ArthurSens** [comment]
I could implement a small shell to get the information:

```bash
#!/bin/bash

dependency="gotest.tools/v3/golden"

# List all packages in the project
packages=$(go list ./...)

# Loop through each package and check if it imports the specific dependency
for pkg in $packages; do
    # Use 'go list' with 'XTestImports' template to get the imports from test binaries
    imports=$(go list -f '{{join .XTestImports "\n"}}' $pkg)
    
    # Check if the dependency is in the imports
    if echo "$imports" | grep -q "$dependency"; then
        # If the dependency is found, print the package path
        echo "$pkg"
    fi
done
```

**Comment by xiu** [comment]
Looks good to me :) Should we add a mention of this all to CONTRIBUTING.md? To guide people into using golden rather than the old way.

**Comment by nicolastakashi** [comment]
Tks @ArthurSens for the shell script, very useful 😄

**Comment by nicolastakashi** [comment]
@xiu I added a entry to the contributing.md but I'm not sure if the information I put is enough?

**Review by ArthurSens — APPROVED** [review]
I'm just not sure about the CONTRIBUTING.md entry, I think we need to have a more elaborated section about testing (and that would include explaining the new Makefile step added in this PR)

I often see PRs with simple mistakes that could easily be caught by running `make test-unit` locally. In another PR, we could probably improve testing guidelines, including documentation for running unit tests, how to update golden files, when to use golden files and even a step-by-step guide for our more complicated e2e tests, how to run all, how to run just a subset of tests, etc.

But of course, out of scope for this PR!

File: scripts/update-golden-files.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
```
Comment (simonpasquier) [inline scripts/update-golden-files.sh:2]: A short explanation of what the script is meant for would be useful.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -61,6 +61,7 @@ This is a rough outline of what a contributor's workflow looks like:
 - Make sure your commit messages are in the proper format (see below).
 - Push your changes to a topic branch in your fork of the repository.
 - Make sure the tests pass, and add any new tests as appropriate.
+- If the tests are checking long strings such as yaml, json or any other complex content, ensure you're using golden files.
```
Comment (simonpasquier) [inline CONTRIBUTING.md:64]:
```suggestion
- If the tests are checking long strings such as YAML, JSON or any other complex content, ensure you're using [golden files](https://pkg.go.dev/gotest.tools/v3/golden).
```

**Review by simonpasquier — COMMENTED** [review]
lgtm, just a few nits!

File: scripts/update-golden-files.sh (hunk 1 above)
Comment (nicolastakashi) [reply scripts/update-golden-files.sh:4]: Can you check again @simonpasquier wdyt?

**Review by simonpasquier — APPROVED** [review]
Thanks!

**Comment by ArthurSens** [comment]
Ah, @nicolastakashi, I just changed the PR description since this change is not relevant for end-users :)

**Pull request merged**